test/e2e/cli/cli_test.go

```diff
@@ -120,8 +121,43 @@ var _ = Describe("cli operation", func() {
 
 	})
 
+	Context("certificate generation per user", func() {
+		It("should have worked, and we can have a certificate", func() {
+			out, err := harness.CLI("certificate", "request", "-n", randString(5))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("certificate is ready"))
+		})
+	})
+
+	Context("list devices", func() {
+		It("Should let you list devices", func() {
+			out, err := harness.CLI("get", "devices")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("Fleet/default"))
+		})
+	})
+
+	Context("list fleets", func() {
+		It("Should let you list fleets", func() {
+			out, err := harness.CLI("get", "fleets")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("e2e-test-fleet"))
+		})
+	})
 })
 
+func randString(n int) string {
+	const alphanum = "abcdefghijklmnopqrstuvwxyz"
+	var bytes = make([]byte, n)
+	if _, err := rand.Read(bytes); err != nil {
+		return "fail to generate username"
+	}
```

Comment:
```suggestion
	Expect(err).ToNot(HaveOccurred())
```

Otherwise if this failed for some reason (this is very unlikely to fail...) we would return
"fail to generate username" and the tests would probably fail with the names with spaces.